distributed/comm/ucx.py

```diff
@@ -39,7 +47,41 @@ def init_once():
     import ucp as _ucp
 
     ucp = _ucp
-    ucp.init(options=dask.config.get("ucx"), env_takes_precedence=True)
+
+    # configuration of UCX can happen in two ways:
+    # 1) high level on/off flags which correspond to UCX configuration
+    # 2) explicity defined UCX configuration flags
+
+    # To handle high level flags we grab the full ucx configuration
+    # pop flags from the dictionary and proceed to check flags.  We do this
+    # because UCX/UCP will not nicely process flags it does not know about
+    # ucp.init(options={'foo': 'bar'}) -- `UCXConfigError`
+    ucx_conf = dask.config.get("ucx")
+
+    enable_infiniband = ucx_conf.pop("infiniband", None)
+    enable_nvlink = ucx_conf.pop("nvlink", None)
+    tcp_over_ucx = ucx_conf.pop("tcp-over-ucx", None)
+    net_devices = ucx_conf.pop("net-devices", None)
```

Comment:
Now that you have included `null` in the yaml file you can safely drop the default values here.  In general we try not to include default values in code, so that the defaults in the yaml file are respected if they are changed in the future.